src/OpenLoco/src/Ui/WindowManager.cpp

```diff
@@ -1466,6 +1467,103 @@ namespace OpenLoco::Ui::WindowManager
         invalidateWidget(window->type, window->number, widgetIndex);
     }
 
+    // Allow mouse wheel scrolling to manipulate stepper widgets and tool sizes
+    static bool stepperWheelInput(Window& w, WidgetIndex_t widgetIndex, int32_t wheel)
+    {
+        auto widgetType = w.widgets[widgetIndex].type;
+
+        // If we aren't scrolling on a tool preview or stepper textbox, lower the widget index
+        // once or twice to try to find the parent widget. This allows scrolling on the
+        // increase/decrease buttons themselves, too.
+        int32_t attempts = 0;
+        while (widgetType != WidgetType::wt_3 && widgetType != WidgetType::textbox && widgetIndex > 0)
+        {
+            switch (widgetType)
+            {
+                case WidgetType::toolbarTab: // + and - for tool widget
+                case WidgetType::button:     // + and - for stepper widget
+                {
+                    if (attempts > 0)
+                    {
+                        // Verify that the previous button was of the same type
+                        auto previousType = w.widgets[widgetIndex + 1].type;
+                        if (previousType != widgetType)
+                        {
+                            return false;
+                        }
+                    }
+                    break;
+                }
+                default:
+                    // The widget type is not an increment or decrement button
+                    return false;
+            }
+
+            attempts++;
+            if (attempts > 2)
+            {
+                // We've looked two widgets up, and no tool preview or stepper widget was found
+                return false;
+            }
+
+            widgetIndex--;
+            widgetType = w.widgets[widgetIndex].type;
+        }
+
+        WidgetIndex_t buttonWidgetIndex;
+        WidgetType expectedType;
+
+        switch (widgetType)
+        {
+            case WidgetType::wt_3:
+            {
+                auto expectedContent1 = Gfx::recolour(ImageIds::decrease_tool_area, Colour::white);
+                auto expectedContent2 = Gfx::recolour(ImageIds::increase_tool_area, Colour::white);
+
+                auto button1Image = w.widgets[widgetIndex + 1].image;
+                auto button2Image = w.widgets[widgetIndex + 2].image;
```

Comment:
its dangerous to access +2 of a widgetIndex. If there are no more widgets this will be past the end.

Reply:
At this point, we know for sure that we're at a stepper widget, though.

Reply:
No i don't think we do. This is the branch for if we are hovering over the land button thingy. The stepper branch was the first loop.

There are basically 3 different parts to this function:
1. Stepper widgets (+-) first loop handles that.
2. wt_3 (land button thingy) part of this switch
3. textbox part of this switch

The first bit handles the steppers for wt_3 and textbox but this switch handles when you are hovering over the text box or wt_3.

Reply:
Ah, sorry for glossing over it, but the terraform size stepper has a similar widget layout, right. It's also preview/lower/higher in sequence.

Reply:
Just add an assert that `w.widgets[widgetIndex + 1].type != WidgetType::end` then we can be confident that nothing goes off the end. As long as that is valid then its safe to access widgetIndex + 2. And if widgetIndex+2 is end that will be caught by the check on L1557.

Reply:
Alright, added assertions for both.